### PR TITLE
ER check examples, more descriptive configuration for `maxClassNameLen`

### PR DESCRIPTION
**Pull request by h1alexbel**
@l3r8yJ take a look, please

---

## PR-Codex overview
This PR focuses on making changes to the codebase related to mutable objects, parsers, validators, controllers, utility classes, and long class names.

### Detailed summary
- Updated the README.md file to fix formatting and provide information about mutable objects and how to suppress checks.
- Made changes to the MyInvalidClass and MyValidClass classes.
- Added examples of valid and invalid classes in the Parsers, Validators, Controllers section of the README.md file.
- Updated the Utility classes section in the README.md file.
- Added an example of a valid class with a long name in the README.md file and provided information on how to configure the maximal reasonable length.

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

### CODE REVIEW

**Comment by zoeself** [comment]
@h1alexbel thank you for your Pull Request. I'll assign someone to review it soon. 

 If this PR solves a ``todo`` from the code, please don't forget to remove it.

**Comment by zoeself** [comment]
@l3r8yJ please review this Pull Request. Deadline (when it should be merged or closed) is ``2024-01-06T07:20:06.419599``.

You should check if the requirements have been implemented (partially or in full), if there are unit tests covering the changes and if the CI build passes. Feel free to reject the PR or ask for changes if it's too big or not clear enough.

Estimation here is ``30 minutes``, that's how much you will be paid. You will be paid even if this PR gets rejected.

File: README.md [hunk 1]
```diff
@@ -128,10 +148,14 @@ class AbstractDatabaseConnection {
...
 ```xml
-<configuration>
-  <maxClassNameLen>15</maxClassNameLen>
+<plugin>
+  <groupId>ru.l3r8y</groupId>
+  <artifactId>oop-cop</artifactId>
   ...
-</configuration>
+  <configuration>
+    <maxClassNameLen>15</maxClassNameLen>
```
Comment (l3r8yJ) [inline README.md:156]: you can add a comment that this is a default value that does not need to be specified

**Review by l3r8yJ — CHANGES_REQUESTED** [review]
@h1alexbel take a look at my comments below, please

File: README.md (hunk 1 above)
Comment (h1alexbel) [reply README.md:156]: you mean `13`?

Comment (l3r8yJ) [reply README.md:156]: > you mean `13`?

sure

**Comment by h1alexbel** [comment]
@l3r8yJ refined, take a look

**Comment by l3r8yJ** [comment]
@h1alexbel seems like `codecov` action has just broken for sometime, i'll rerun it a bit later

**Comment by l3r8yJ** [comment]
@h1alexbel can you, please, also add a info about new suppressions?

**Comment by h1alexbel** [comment]
@l3r8yJ already added

**Comment by zoeself** [comment]
@l3r8yJ thank you for resolving this ticket. I've just added it to your active invoice. You can always check all your invoices and more on the [Contributor Dashboard](https://self-xdsd.com).

**Pull request merged**